papas/reconstruction/src/PFBlock.cpp

```diff
@@ -162,28 +169,34 @@ std::string PFBlock::edgeMatrixString() const {
         if (e1 == e2) {
           out.write("       .");  // diagonal
           break;
-        } else if (edge(e1, e2).distance() < 0)
-          out.write("     ---");  //-ve distance
-        else if (edge(e1, e2).isLinked() == false)
-          out.write("     xxx");  // not linked
-        else {                    // linked and has distance
-          out.write("{:8.4f}", edge(e1, e2).distance());
+        } else if (hasEdge(e1, e2)) {
+          const auto& ed = edge(e1, e2);
```

Comment:
just a detail but: in that case I would go for the explicit "edge" as a variable name rather that "ed"